src/Controller/SearchEngineController.php

```diff
@@ -72,56 +141,146 @@ public function generate_index()
-        
-        // Use the generated index in the previous step
-        $tnt->selectIndex('groups.index');
-        
-        $maxResults = 20;
-        
-        $this->setFuzziness($tnt);
-
-        // Search for a band named like "Guns n' roses"
-        $results = $tnt->search("que", $maxResults);
-        
-        // Keep a reference to the Doctrine repository of artists
-        $usergroupsRepository = $em->getRepository(Usergroup::class);
-        
-        // Store the results in an array
-        $rows = [];
-        
-        foreach($results["ids"] as $id){
-            // You can optimize this by using the FIELD function of MySQL if you are using mysql
-            // more info at: https://ourcodeworld.com/articles/read/1162/how-to-order-a-doctrine-2-query-result-by-a-specific-order-of-an-array-using-mysql-in-symfony-5
-            $userGroup = $usergroupsRepository->find($id);
-            
-            $rows[] = [
-                'id' => $userGroup->getId(),
-                'name' => $userGroup->getName(), 
-                'description' => $userGroup->getDescription()
-            ];
-        }
-        
-        // Return the results to the user
-        return new JsonResponse($rows);
+        $tnt->loadConfig($this->getTNTSearchConfiguration());
+		$rows = [];
+		$this->setFuzziness($tnt);
+
+		if( in_array( 'discussions', $categories ) ){
+			$rows['discussions'] = $this->searchResultByCategory(
+				'discussions',
+				self::DISCUSSION_INDEX,
+				DiscussionMessage::class,
+				['id', 'title', 'body', 'author', 'group'],
+				$tnt,
+				$text,
+				$em
+			);
+		} else {
+			$rows['discussions'] = [];
+		}
+
+		if( in_array( 'actualites', $categories ) ){
+			$rows['actualites'] = $this->searchResultByCategory(
+				'actualites',
+				self::ACTUALITE_INDEX,
+				Article::class,
+				['id', 'title', 'body', 'author', 'group'],
+				$tnt,
+				$text,
+				$em
+			);
+		} else {
+			$rows['actualites'] = [];
+		}
+
+		if( in_array( 'pages', $categories ) ){
+			$rows['pages'] = $this->searchResultByCategory(
+				'pages',
+				self::PAGE_INDEX,
+				Page::class,
+				['id', 'title', 'body', 'author', 'group'],
+				$tnt,
+				$text,
+				$em
+			);
+		} else {
+			$rows['pages'] = [];
+		}
+
+		if( in_array( 'documents', $categories ) ){
+			$rows['documents'] = $this->searchResultByCategory(
+				'documents',
+				self::DOCUMENT_INDEX,
+				Document::class,
+				['id', 'title', 'group'],
+				$tnt,
+				$text,
+				$em
+			);
+		} else {
+			$rows['documents'] = [];
+		}
+
+		if( in_array( 'membres', $categories ) ){
+			$rows['membres'] = $this->searchResultByCategory(
+				'membres',
+				self::MEMBER_INDEX,
+				User::class,
+				['id', 'name', 'presentation', 'bio'],
+				$tnt,
+				$text,
+				$em
+			);
+		} else {
+			$rows['membres'] = [];
+		}
+
+        return $rows;
     }
 
+	private function searchResultByCategory($category, $index, $class, $propertyList, $tnt, $text, $em){
```

Comment:
```suggestion
	private function searchResultByCategory($category, $index, $repository, $propertyList, $tnt, $text){
```
Tu peux passer le repository adequat directement